tests/MongoDB.Driver.Tests/ReadPreferenceOnStandaloneTests.cs

```diff
@@ -0,0 +1,87 @@
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using FluentAssertions;
+using MongoDB.Bson;
+using MongoDB.Bson.TestHelpers.XunitExtensions;
+using MongoDB.Driver.Core;
+using MongoDB.Driver.Core.Clusters;
+using MongoDB.Driver.Core.Events;
+using MongoDB.Driver.Core.Misc;
+using MongoDB.Driver.TestHelpers;
+using Xunit;
+
+namespace MongoDB.Driver.Tests
+{
+    public class ReadPreferenceOnStandaloneTests
+    {
+        [Theory]
+        [ParameterAttributeData]
+        public void ReadPreference_should_not_be_sent_to_standalone_server(
+            [Values(false, true)] bool async)
+        {
+            var eventCapturer = new EventCapturer().Capture<CommandStartedEvent>(e => e.CommandName.Equals("find") || e.CommandName.Equals("$query"));
+            using (var client = CreateDisposableClient(eventCapturer, ReadPreference.PrimaryPreferred))
+            {
+                var database = client.GetDatabase(DriverTestConfiguration.DatabaseNamespace.DatabaseName);
+                var collection = database.GetCollection<BsonDocument>(DriverTestConfiguration.CollectionNamespace.CollectionName);
+
+                if (async)
+                {
+                    var _ = collection.FindAsync("{ x : 2 }").GetAwaiter().GetResult();
+                }
+                else
+                {
+                    var _ = collection.FindSync("{ x : 2 }");
+                }
+
+                CommandStartedEvent sentCommand = ((CommandStartedEvent)eventCapturer.Events[0]);
+
+                var serverVersion = client.Cluster.Description.Servers[0].Version;
+
+                if (client.Cluster.Description.Type == ClusterType.Standalone)
+                {
+                    sentCommand.Command.Contains("readPreference").Should().BeFalse();
+                }
+                else if (client.Cluster.Description.Type == ClusterType.Sharded &&
+                         serverVersion < Feature.CommandMessage.FirstSupportedVersion)
+                {
+                    if (sentCommand.CommandName.Equals("$query"))
+                    {
+                        sentCommand.Command.Contains("$readPreference").Should().BeTrue();
+                    }
+                    else if (sentCommand.CommandName.Equals("find"))
+                    {
+                        sentCommand.Command.Contains("readPreference").Should().BeTrue();
+                    }
+                }
+                else if (serverVersion >= Feature.CommandMessage.FirstSupportedVersion)
+                {
+                    sentCommand.Command.Contains("$readPreference").Should().BeTrue();
+                }
+            }
```

Comment:
I think lines 54-73 can be simplified to:

    var clusterType = client.Cluster.Description.Type;
    var expectedContainsReadPreference = clusterType == ClusterType.Standalone ? false : true;
    sentCommand.Command.Contains("$readPreference").Should().Be(expectedContainsReadPreference);

Reply:
Done (with modifications)